### PR TITLE
Nft

### PR DESCRIPTION
**Pull request by Ic3Sandy**

### CODE REVIEW

File: lib/merchant-wallet.ts [hunk 1]
```diff
@@ -300,4 +301,118 @@ export class MerchantWallet {
     });
     return tx;
   }
+
+  async safeTransferFromERC721(
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts:304]: Do we need to the return type here?

File: lib/merchant-wallet.ts [hunk 2]
```diff
@@ -300,4 +301,118 @@ export class MerchantWallet {
...
+    erc721Address: string,
+    from: string,
+    to: string,
+    tokenId: number
+  ) {
+    const contract = new this.client.Contract(AbiERC721, erc721Address);
+    const encodeData = contract.methods
+      .safeTransferFrom(from, to, tokenId)
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts]: Do `safeTransferFrom` accept data as the forth field?

File: lib/merchant-wallet.ts [hunk 3]
```diff
@@ -300,4 +301,118 @@ export class MerchantWallet {
...
+      data: encodeData,
+    });
+  }
+
+  async balanceOfERC721(erc721Address: string): Promise<number> {
+    const contract = new this.client.Contract(AbiERC721, erc721Address);
+    return Number(
+      (await contract.methods.balanceOf(await this.address()).call())["0"]
+    );
+  }
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts:478]: Missign the field `getApprovedERC721`?

Comment (thebevrishot) [reply lib/merchant-wallet.ts:478]: `isApprovedForAllERC721` also?

File: lib/merchant-wallet.ts [hunk 4]
```diff
@@ -300,4 +301,118 @@ export class MerchantWallet {
...
+      to: erc721Address,
+      data: encodeData,
+    });
+  }
+
+  async safeBatchTransferFromERC1155(
+    erc1155Address: string,
+    from: string,
+    to: string,
+    tokenIds: number[],
```
Comment (thebevrishot) [inline lib/merchant-wallet.ts]: String?

**Review by thebevrishot — CHANGES_REQUESTED** [review]
Seem like we missing some methods from standard 
`https://eips.ethereum.org/EIPS/eip-1155`
and `https://eips.ethereum.org/EIPS/eip-721`

And we could not use number to represent uint256 for solidity since it's exceed size of safe integer(It would be convert to float)

File: lib/merchant-wallet.ts (hunk 1 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts:304]: Ok krub, I will add the return type.

File: lib/merchant-wallet.ts (hunk 2 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts]: yes, I will add data with default value for the forth field

File: lib/merchant-wallet.ts (hunk 3 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts:478]: I didn't add those method because I think it not relate with merchant wallet

Comment (Ic3Sandy) [reply lib/merchant-wallet.ts:478]: `function getApproved(uint256 _tokenId) external view returns (address);`

It receive tokenId I not sure to add function like this to merchant wallet

File: lib/merchant-wallet.ts (hunk 4 above)
Comment (Ic3Sandy) [reply lib/merchant-wallet.ts]: I see krub, I will change it to string krub

**Comment by tanakornon** [comment]
LGTM

**Pull request merged**